src/runtime_src/core/common/device.h

```diff
@@ -511,6 +511,28 @@ device_query(const std::shared_ptr<device>& device, Args&&... args)
   return boost::any_cast<typename QueryRequestType::result_type>(ret);
 }
 
+template <typename QueryRequestType>
+inline typename QueryRequestType::result_type
+device_query_default(const device* device, const typename QueryRequestType::result_type& default_value)
+{
+  try {
+    return device_query<QueryRequestType>(device);
+  } catch(...) {
+    return default_value;
+  }
+}
+
+template <typename QueryRequestType>
+inline typename QueryRequestType::result_type
+device_query_default(const std::shared_ptr<device>& device, const typename QueryRequestType::result_type& default_value)
+{
+  try {
+    return device_query<QueryRequestType>(device);
+  } catch(...) {
```

Comment:
```suggestion
  } 
  catch(const query::exception&) {
```
I think you need to do one more change to make this work properly.  I don't want device.h to #include query_requests.h to get at the exception class.  I think the exception class should be moved to query.h which is already included in device.h